reana-ui/src/components/InteractiveSessionModal.js

```diff
@@ -0,0 +1,194 @@
+  );
+
+  const isCustomAllowed = environments[sessionType].allow_custom;
+
+  const checkFields = () => {
+    const errors = {};
+    if (environmentType === EnvironmentType.Custom && !customImage) {
+      errors.customImage = "Please provide a custom image";
+    }
+    setErrors(errors);
+    return Object.keys(errors).length === 0;
+  };
+
+  const onCloseModal = () => {
+    resetForm();
+    dispatch(closeInteractiveSessionModal());
+  };
+
+  const onSubmit = () => {
+    if (!checkFields()) return;
+
+    setIsLoading(true);
+
+    const image =
+      environmentType === EnvironmentType.Recommended
+        ? recommendedImage
+        : customImage;
+    dispatch(
+      openInteractiveSession(workflow.id, { type: sessionType, image }),
+    ).finally(onCloseModal);
+  };
+
+  return (
+    <Modal open={open} onClose={onCloseModal} closeIcon size="small">
+      <Modal.Header>Open interactive session</Modal.Header>
+      <Modal.Content>
+        <Form id="formOpenSession" onSubmit={onSubmit} loading={isLoading}>
+          {/* only show dropdown if there are at least two session types to choose from */}
+          {allSessionTypes.length > 1 && (
+            <FormDropdown
+              selection
+              label="Session type"
+              options={sessionTypeOptions}
+              value={sessionType}
+              onChange={(_, { value }) => resetForm(value)}
+              disabled={sessionTypeOptions.length === 1}
+            />
+          )}
+          {/* show selector only if custom images are allowed */}
+          {isCustomAllowed && (
+            <FormGroup inline>
+              <label>Environment</label>
+              <FormRadio
+                name="environmentType"
+                label="Recommended environments"
+                value="recommended"
+                checked={environmentType === EnvironmentType.Recommended}
+                onChange={(_, { value }) => setEnvironmentType(value)}
+              />
+              <FormRadio
+                name="environmentType"
+                label="Custom environment"
+                value="custom"
+                checked={environmentType === EnvironmentType.Custom}
+                onChange={(_, { value }) => setEnvironmentType(value)}
+              />
+            </FormGroup>
+          )}
+          {environmentType === EnvironmentType.Recommended && (
+            <FormDropdown
+              selection
+              label="Recommended environments"
+              options={recommendedImageOptions}
+              value={recommendedImage}
+              onChange={(_, { value }) => setRecommendedImage(value)}
+              search={true}
+            />
+          )}
+          {environmentType === EnvironmentType.Custom && (
+            <FormInput
+              label="Custom environment"
+              value={customImage}
+              onChange={(_, { value }) => setCustomImage(value)}
+              placeholder={"Custom container image"}
```

Comment:
When custom environment images are enabled, the user may try to type fancy values such as:

```
alert("XSS")
```

This leads to a success message:

```
Success!

The interactive session has been created. However, it could take several minutes to start the Jupyter Notebook. Click on the Jupyter logo to access it. 
```

And on the backend, the session pod is created but obviously not started due to the `InvalidImageName` error.

Could we plug some value sanitisation checks so that we would prevent even trying to open invalid image strings?

(Ideally on the backend side, so that we would also cover `reana-client open`, but we could also eliminate some characters already in the front-end perhaps, such as in the various XSS test strings, as tried above.)

Reply:
I think we should do this as part of another issue/PR, as these checks could be also done for example for the docker images specified in `reana.yaml`